src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Members.cs

```diff
@@ -609,7 +609,17 @@ void visitFunctionPointerSignature(IMethodSymbol symbol)
 
                 foreach (var param in symbol.Parameters)
                 {
-                    param.Accept(this.NotFirstVisitor);
+                    if (format.MemberOptions.IncludesOption(SymbolDisplayMemberOptions.IncludeRef))
```

Comment:
>if (format.MemberOptions.IncludesOption(SymbolDisplayMemberOptions.IncludeRef)) [](start = 20, length = 79)

It looks like this breaks some tests because diagnostics now doesn't display ref-ness of parameter types by default. Perhaps we should display ref-ness in function pointer types unconditionally (including return ref-ness)?

Reply:
I'm ok with that

Reply:
@333fred 

> Perhaps we should display ref-ness in function pointer types unconditionally (including return ref-ness)?

What do you think? Another option is to add a whole new display option (SymbolDisplayMiscellaneousOptions?) that controls that for function pointer types.

---
In reply to: [592331577](https://github.com/dotnet/roslyn/pull/51223#discussion_r592331577) [](ancestors = 592331577)

Reply:
It looks like all the remaining broken tests are due to this issue. I'm going to wait to fix them till a decision is made.

Reply:
> Perhaps we should display ref-ness in function pointer types unconditionally (including return ref-ness)

I think we should do this. You can overload methods by function-pointer-parameter-refness, so if you had `M(delegate*<ref string, void>)` and `M(delegate*<string, void>)`, that would lead to bad diagnostics.

Reply:
> It looks like all the remaining broken tests are due to this issue. I'm going to wait to fix them till a decision is made.

It looks like we are converging on displaying ref-ness in function pointer types unconditionally (including the return ref-ness)

---
In reply to: [592505606](https://github.com/dotnet/roslyn/pull/51223#discussion_r592505606) [](ancestors = 592505606)